easybuild/easyblocks/b/boost.py

```diff
@@ -193,6 +194,36 @@ def build_step(self):
             
             # cleanup previous build before proceeding with the full Boost
             run_cmd("./bjam --clean-all", log_all=True, simple=True)
+        
+        if self.cfg['boost_multi_thread']:
+            self.log.info("Building boost with multi threading")
+
+            bjammtoptions = "%s threading=multi --layout=tagged" % bjamoptions
+
+            # build multi-thread lib
+            run_cmd("./bjam %s %s" % (bjammtoptions, paracmd), log_all=True, simple=True)
+
+            # install the multi-thread version of boost
+            run_cmd("./bjam %s  install %s" % (bjammtoptions, paracmd), log_all=True, simple=True)
+            
+            # cleanup previous build before proceeding with the full Boost
+            run_cmd("./bjam --clean-all", log_all=True, simple=True)
+
+        # if both boost_mpi and boost_multi_thread are enabled, build boost mpi with multi-thread support
+        if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
+            self.log.info("Building boost_mpi with multi threading")
+
+            bjammpimtoptions = "%s  --user-config=user-config.jam --with-mpi threading=multi --layout=tagged" % bjamoptions
+
+            # build mpi multi-thread lib
+            run_cmd("./bjam %s %s" % (bjammpimtoptions, paracmd), log_all=True, simple=True)
+
+            # install the multi-thread version of boost
+            run_cmd("./bjam %s  install %s" % (bjammpimtoptions, paracmd), log_all=True, simple=True)
+            
+            # cleanup previous build before proceeding with the full Boost
+            run_cmd("./bjam --clean-all", log_all=True, simple=True)
```

Comment:
@mboisson This is a bit too copy-pasty...

We should define a separate method and reuse that (note: untested code below):

```python

def build_boost_variant(self, bjamoptions):
    """Build Boost library with specified options for bjam."""
    # build with specified options
    run_cmd("./bjam %s %s" % (bjamoptions, paracmd), log_all=True, simple=True)
    # install built Boost library
    run_cmd("./bjam %s install %s" % (bjamoptions, paracmd), log_all=True, simple=True)
    # clean up before proceeding with next build
    run_cmd("./bjam --clean-all", log_all=True, simple=True)

def build_step(self):
    ...
    if self.cfg['boost_mpi']:
        self.log.info("Building Boost MPI library")
        self.build_boost_variant(bjamoptions + "--user-config=user-config.jam --with-mpi")

    if self.cfg['boost_multi_thread']:
        self.log.info("Building Boost multi-threading library")
        self.build_boost_variant(bjamoptions + "threading=multi --layout=tagged")

    if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
        self.log.info("Building Boost MPI library with multi-threading")
        self.build_boost_variant(bjamoptions + "--user-config=user-config.jam --with-mpi threading=multi --layout=tagged")
```